File: front-office/src/app/services/trip-alert.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CreateTripAlertRequest, TripAlert } from '../models/trip-alert.model';

@Injectable({
  providedIn: 'root',
})
export class TripAlertService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = '/api/trip-alerts';

  createAlert(request: CreateTripAlertRequest): Observable<TripAlert> {
    return this.http.post<TripAlert>(this.baseUrl, request);
  }

  getMyAlerts(): Observable<TripAlert[]> {
    return this.http.get<TripAlert[]>(`${this.baseUrl}/mine`);
  }

  deleteAlert(alertId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${alertId}`);
  }
}

File: back-office/src/main/java/com/coliclic/backoffice/tripalert/service/TripAlertService.java
package com.coliclic.backoffice.tripalert.service;

import com.coliclic.backoffice.trip.entity.Trip;
import com.coliclic.backoffice.tripalert.dto.CreateTripAlertRequest;
import com.coliclic.backoffice.tripalert.dto.TripAlertResponse;
import com.coliclic.backoffice.user.entity.User;

import java.util.List;

public interface TripAlertService {

    TripAlertResponse createAlert(CreateTripAlertRequest request, User user);

    List<TripAlertResponse> getMyAlerts(User user);

    void deleteAlert(Long alertId, User user);

    void notifyMatchingAlerts(Trip trip);
}
